tools/cdf-lineage-exporter/cdap_lineage_export/cdap/client.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2019 Google LLC.
+# This software is provided as-is, without warranty or representation
+# for any use or purpose.
+# Your use of it is subject to your agreement with Google.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import urllib.parse
+from dataclasses import dataclass
+from functools import wraps
+
+from google.api_core.exceptions import NotFound
+from google.auth.credentials import AnonymousCredentials, Credentials
+from google.cloud import _http
+from google.cloud.client import ClientWithProject
```

Comment:
I really think it's unnecessary to use the `google-cloud-core` library for this client. That package is predominantly used by GCS and BigQuery, both of wish have a desire to eventually move to gRPC (not sure on the roadmap, though)

I'll send a PR to your fork with how I'd suggest structuring this.

Reply:
FYI I have admin rights on this repo so I was lazy and just created a branch.
You can open PR against feature/cdapLineageExporter not my fork.